MacOS 14 strength.cpp compilation error

The following error during `pip install OApackage`

`clang --version `
```                                    
Apple clang version 16.0.0 (clang-1600.0.26.6)
Target: x86_64-apple-darwin24.2.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```
Error log:
```
clang++ -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk -I. -Isrc -I/private/var/folders/hy/p_qw44dj5d1f4j79cf4m_10h0000gn/T/pip-build-env-4vqha2an/overlay/lib/python3.10/site-packages/numpy/_core/include -I/Users/myuser/Work/GitHub/test-project/oa/.venv/include -I/usr/local/opt/python@3.10/Frameworks/Python.framework/Versions/3.10/include/python3.10 -c src/strength.cpp -o build/temp.macosx-14-x86_64-cpython-310/src/strength.o -DSWIGCODE -DFULLPACKAGE -Isrc/nauty/ -DNOOMP -DUSEZLIB -O3 -Wno-unknown-pragmas -Wno-sign-compare -Wno-return-type -Wno-unused-variable -Wno-unused-result -fPIC -Wno-date-time
      In file included from src/strength.cpp:15:
      In file included from src/strength.h:15:
      src/extend.h:72:58: warning: use of enumeration in a nested name specifier is a C++11 extension [-Wc++11-extensions]
         72 |               init_column_previous (1), extendarraymode (extendarray_mode_t::APPENDFULL), j5structure (J5_45), algmode (MODE_AUTOSELECT){};
            |                                                          ^
      src/extend.h:93:58: warning: use of enumeration in a nested name specifier is a C++11 extension [-Wc++11-extensions]
         93 |               init_column_previous (1), extendarraymode (extendarray_mode_t::APPENDFULL), j5structure (J5_45), algmode (MODE_AUTOSELECT) {
            |                                                          ^
      src/extend.h:257:36: warning: use of enumeration in a nested name specifier is a C++11 extension [-Wc++11-extensions]
        257 |         dextend_t () : filtermode (dfilter_t::DFILTER_MULTI), Dcheck (dcalc_mode::DCALC_COND), directcheck (1){};
            |                                    ^
      src/extend.h:257:71: warning: use of enumeration in a nested name specifier is a C++11 extension [-Wc++11-extensions]
        257 |         dextend_t () : filtermode (dfilter_t::DFILTER_MULTI), Dcheck (dcalc_mode::DCALC_COND), directcheck (1){};
            |                                                                       ^
      src/strength.cpp:246:47: error: expected ';' after expression
        246 |         tmp = new int[ncombs(ncols, strength)] {};
            |                                               ^
            |                                               ;
      4 warnings and 1 error generated.
      error: command '/usr/bin/clang++' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for OApackage
```

--- a/src/arrayproperties.cpp
+++ b/src/arrayproperties.cpp
@@ -355,10 +355,10 @@ std::vector< double > projDeff (const array_link &orthogonal_array, int number_o
         int m = 1 + number_of_factors + number_of_factors * (number_of_factors - 1) / 2;
         int N = orthogonal_array.n_rows;
 
-        if (verbose)
+        if (verbose) {
                 myprintf ("projDeff: k %d, kp %d: start with %ld combinations \n", number_of_columns,
                           number_of_factors, (long)number_combinations);
-
+        }
 		std::vector< double > efficiencies(number_combinations);
 		for (int64_t i = 0; i < number_combinations; i++) {
                 if (m > N)

--- a/src/arraytools.cpp
+++ b/src/arraytools.cpp
@@ -2373,7 +2373,7 @@ std::vector< int > numberModelParams(const array_link &array, int order)
 		arraydata_t arrayclass = arraylink2arraydata(array, 0, 0);
 		std::vector< int > s = arrayclass.factor_levels();
 		std::vector< int > df = s;
-		std::transform(df.begin(), df.end(), df.begin(), std::bind2nd(std::minus< int >(), 1.0));
+		std::transform(df.begin(), df.end(), df.begin(), std::bind(std::minus< int >(), std::placeholders::_1, 1.0));
 
 		/* main effect contrasts */
 		int mesize = std::accumulate(df.begin(), df.end(), 0);
@@ -2542,7 +2542,7 @@ std::pair< MatrixFloat, MatrixFloat > array2eigenModelMatrixMixed (const array_l
         std::vector< int > s = arrayclass.factor_levels ();
 
         std::vector< int > df = s;
-        std::transform (df.begin (), df.end (), df.begin (), std::bind2nd (std::minus< int > (), 1.0));
+        std::transform (df.begin (), df.end (), df.begin (), std::bind (std::minus< int > (), std::placeholders::_1, 1.0));
 
         if (verbose >= 2) {
                 arrayclass.show ();

--- a/src/nonroot.cpp
+++ b/src/nonroot.cpp
@@ -644,11 +644,6 @@ inline lmc_t LMC_check_col_j5order (const array_t *original, const array_t *arra
                 ret = LMC_check_col (original + dd->col * +ad->N, array + cpoffset, lperm, ad, dd);
                 return ret;
         } else {
-                if (log_print (DEBUG, "")) {
-                        myprintf ("  j5order: col %d, jbase %d, jcol %d: colperm: ", dd->col, jbase, jcol);
-                        print_perm (pp, 5);
-                }
-
                 if (jbase ORDER_J5_SMALLER jcol)
                         ret = LMC_MORE;
                 else

--- a/src/strength.cpp
+++ b/src/strength.cpp
@@ -243,7 +243,7 @@ rev_index *create_reverse_colcombs (colindex_t **colcombs, const int ncols, cons
         const int ncolcombs = ncombs (ncols, strength);
 
         rev_colcombs = (rev_index *)malloc (ncols * sizeof (rev_index));
-        tmp = new int[ncombs(ncols, strength)] {};
+        tmp = new int[ncombs(ncols, strength)] ();
 
         log_print (DEBUG, "ncols = %i\n", ncols);
 

--- a/src/version.h
+++ b/src/version.h
@@ -1,1 +1,1 @@
-const char *__version__ = "2.7.14";
+const char *__version__ = "2.7.15";

